lib/spack/spack/cmd/commands.py

```diff
@@ -262,20 +262,7 @@ def _commands(parser, args):
     if args.header and not os.path.exists(args.header):
         tty.die("No such file: '%s'" % args.header)
 
-    # if we're updating an existing file, only write output if a command
-    # or the header is newer than the file.
     if args.update:
-        if os.path.exists(args.update):
-            files = [
-                spack.cmd.get_module(command).__file__.rstrip('c')  # pyc -> py
-                for command in spack.cmd.all_commands()]
-            if args.header:
-                files.append(args.header)
-            last_update = os.path.getmtime(args.update)
-            if not any(os.path.getmtime(f) > last_update for f in files):
-                tty.msg('File is up to date: %s' % args.update)
-                return
```

Comment:
This logic doesn't work when a command is removed, since we can't check if a non-existent file is newer than the file to update.